### DISCUSSION TITLE
[array] array destructuring

### DISCUSSION THREAD

**Discussion by igotfr** [vlang/v#13641] [Ideas]
```v
[a, b, c] := [1 2 3]
```

**Comment by igotfr** [comment]
related:
https://github.com/vlang/rfcs/issues/19

**Comment by dumblob** [comment]
Any strong motivational real-world use cases which can't be easily covered by the current *V* functionality (e.g. multi-return from functions etc.)?

**Comment by tomByrer** [comment]
> real-world use cases

Well, it is more terse than a function?  Also maps better to [JavaScript's destructuing](https://exploringjs.com/es6/ch_destructuring.html#_array-destructuring).

**Reply by dumblob** [reply]
Yes, it is terser. Though I personally don't find the difference that much convincing - especially considering anonymous functions or closures *V* will (or already does - correct me if I'm wrong) support.

Note also the linked JS destructuring documentation says the primary motivation for using destructuring in JS is to simulate multi-return from functions :wink: - but *V* already supports multi-return **natively**, so therefore I thought more convincing examples could clarify what *V* shortcoming array destructuring would alleviate.
